js/src/logger.ts

```diff
@@ -1036,11 +1034,23 @@ class BackgroundLogger {
   private triggerActiveFlush() {
     if (this.activeFlushResolved) {
       this.activeFlushResolved = false;
-      this.activeFlush = this.flushOnce().finally(() => {
```

Comment:
For some reason, changing the logger stuff uncovered this "uncaught promise rejection" issue, where we assign `this.activeFlush` to a promise which has no `catch` clause. Even though we are guaranteed to throw and catch it later, I guess javascript requires you to attach a catch handler upon promise construction.

So to work around this, we always install a `catch` handler which stores the error as an instance variable, and then we throw it from the instance variable.